src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -3046,33 +3043,22 @@ def groupby_agg(
             SnowflakeQueryCompiler: with a newly constructed internal dataframe
         """
         level = groupby_kwargs.get("level", None)
-        is_supported = check_is_groupby_supported_by_snowflake(
-            by, level, axis
-        ) and check_is_aggregation_supported_in_snowflake(agg_func, agg_kwargs, axis)
 
-        if not is_supported:
-            if agg_func in ["head", "tail"]:
-                # head and tail cannot be run per column - it is run on the
-                # whole table at once.
-                return self._groupby_head_tail(
-                    n=agg_kwargs.get("n", 5),
-                    op_type=agg_func,
-                    by=by,
-                    level=level,
-                    dropna=agg_kwargs.get("dropna", True),
-                )
-            else:
-                # Named aggregates are passed in via agg_kwargs. We should not pass `agg_func` since we have modified
-                # it to be of the form {column_name: (agg_func, new_column_name), ...}, which will cause the error message
-                # to be misformatted. Instead, we re-format into the form f"agg(**named_aggregations)" so that the error message
-                # is recognizable to the user.
-                if using_named_aggregations_for_func(agg_func):
-                    agg_func = format_kwargs_for_error_message(agg_kwargs)
-                    agg_func = f"agg({agg_func})"
+        if agg_func in ["head", "tail"]:
+            # head and tail cannot be run per column - it is run on the
+            # whole table at once.
+            return self._groupby_head_tail(
+                n=agg_kwargs.get("n", 5),
+                op_type=agg_func,
+                by=by,
+                level=level,
+                dropna=agg_kwargs.get("dropna", True),
+            )
 
-                ErrorMessage.not_implemented(
-                    f"Snowpark pandas GroupBy.{agg_func} does not yet support pd.Grouper, axis == 1, by != None and level != None, by containing any non-pandas hashable labels, or unsupported aggregation parameters."
-                )
+        if not check_is_aggregation_supported_in_snowflake(agg_func, agg_kwargs, axis):
```

Comment:
Should the supported check be moved back above head/tail? This might change aggregation behavior for head/tail with certain kwargs.

Reply:
Currently, `head` and `tail` always fall into `if agg_func in ["head", "tail"]` block because `check_is_groupby_supported_by_snowflake` returns `False`. This version of the code is clearer and shouldn't change any behavior for `head` and `tail`.